fn/io_test.go

```diff
@@ -0,0 +1,84 @@
+package fn_test
+
+import (
+	"os"
+	"testing"
+
+	"github.com/lightningnetwork/lnd/fn"
+)
+
+var (
+	data1 = "The network is robust in its unstructured simplicity."
+	data2 = "Programming today is a race between software engineers " +
+		"striving to build bigger and better idiot-proof programs, " +
+		"and the Universe trying to produce bigger and better " +
+		"idiots. So far, the Universe is winning."
+)
+
+// TestWriteFile uses same scenario of ioutil asserting the content created and
+// stored on new file with the original one.
+func TestWriteFile(t *testing.T) {
+	t.Parallel()
+
+	f, deferred := ensureTempfile(t)
+	filename := f.Name()
+	defer deferred()
+
+	// Write file.
+	if err := fn.WriteFile(
+		filename, []byte(data2), 0644, fn.KeepFileOnErr,
+	); err != nil {
```

Comment:
nit: new code should use the `require` library. See the `fn/slice_test.go` test files for examples (this here should be `require.NoError(t, err)` for example).

Reply:
replaced asserts to use require